sympy/stats/frv_types.py

```diff
@@ -401,11 +494,15 @@ def Hypergeometric(name, N, m, n):
 
 
 class RademacherDistribution(SingleFiniteDistribution):
+
     @property
-    @cacheit
-    def dict(self):
-        return {-1: S.Half, 1: S.Half}
+    def set(self):
+        return [-1, 1]
 
+    @property
+    def pdf(self):
+        k = Dummy('k')
+        return Lambda(k, Piecewise((S.Half, Or(Eq(k, -1), Eq(k, 1))), (0, True)))
```

Comment:
can we save the `.dict` method by moving `pdf` to the superclass?

Reply:
I can figure out two things, either you want the `.dict` method to be cached or may be some misinterpretation happened and you thought that `.dict` method is removed from the distributions.

For the first point, `dict` is being cached using `@cacheit` decorator and when we call `density(X).dict`(for large values of dimension) for the second or third times, it gives very quickly, so I believe that caching is working properly.

For the second point, `dict` method is not removed. It is still there in the superclass. Now, why I did so? Because, the `dict` method has a common computational logic for all the distributions, however, that's not the case for `pdf`. Each distribution has their own pdf. In the previous design, the pdf was generated using `dict` hence slowing down the computation. But now since, it is transformed to `Lambda` involving symbolic computation of `Sum` it will take constant time for a single value, which was not the case before. In fact, moving `pdf` to the super class will restrict us to not use symbolic dimension.

An Example:

```python
>>> from sympy.stats import Die, density
>>> from sympy import symbols
>>> D = Die('X', 6)
>>> density(D)
DieDistribution(6)
>>> density(D).dict
{1: 1/6, 2: 1/6, 3: 1/6, 4: 1/6, 5: 1/6, 6: 1/6}
>>> n = symbols('n')
>>> DS = Die('XS', n)
>>> density(DS)
DieDistribution(n)
>>> density(DS).dict
Density(DieDistribution(n))
>>> x = symbols('x')
>>> density(DS)(x)
Piecewise((1/n, (n >= x) & (x >= 1)), (0, True))
>>> density(DS)(1)
Piecewise((1/n, n >= 1), (0, True))
>>> density(DS).dict.subs(n, 10)
Density(DieDistribution(10))
>>> density(DS).dict.subs(n, 10).doit()
{1: 1/10, 2: 1/10, 3: 1/10, 4: 1/10, 5: 1/10, 6: 1/10, 7: 1/10, 8: 1/10, 9: 1/10, 10: 1/10}
```  

Note : This PR doesn't break the current API as no pre-existing tests were modified, only some new tests have been added which work properly.

Please let me know if you wanted something else. Thanks.

Reply:
It looks good.